bun/starter/src/main.ts

```diff
@@ -1,29 +1,31 @@
-import { Client } from "node-appwrite";
+import { Client, Users } from "node-appwrite";
 
-// This is your Appwrite function
-// It's executed each time we get a request
+// This Appwrite function will be executed every time your function is triggered
 export default async ({ req, res, log, error }: any) => {
-  // Why not try the Appwrite SDK?
-  //
-  // const client = new Client()
-  //    .setEndpoint('https://cloud.appwrite.io/v1')
-  //    .setProject(Bun.env["APPWRITE_FUNCTION_PROJECT_ID"])
-  //    .setKey(Bun.env["APPWRITE_API_KEY"]);
+  // You can use the Appwrite SDK to interact with other services
+  // For this example, we're using the Auth service
+  const client = new Client()
+    .setEndpoint(Bun.env["APPWRITE_FUNCTION_API_ENDPOINT"])
+    .setProject(Bun.env["APPWRITE_FUNCTION_PROJECT_ID"])
+    .setKey(req.headers['x-appwrite-key'] ?? '');
+  const users = new Users(client);
 
-  // You can log messages to the console
-  log("Hello, Logs!");
-
-  // If something goes wrong, log an error
-  error("Hello, Errors!");
+  try {
+    const response = await users.list();
+    // Log messages and errors to the Appwrite Console
+    // These logs won't be seen by your end users
+    log(`Amount of users: ${response.total}`);
+  } catch(err) {
+    error("Could not list users: " + err.message);
+  }
 
   // The `req` object contains the request data
-  if (req.method === "GET") {
-    // Send a response with the res object helpers
-    // `res.send()` dispatches a string back to the client
-    return res.send("Hello, World!");
+  if (req.path === "/ping") {
+    // Use `res` object to respond with `text()`, `json()`, or `binary()`
+    // Don't forget to return a response!
+    return res.text("Pong");
   }
 
-  // `res.json()` is a handy helper for sending JSON
```

Comment:
any reason to remove this comment? It is still relevant

Reply:
It was mentioned earlier you can do json() on res. I removed this to keep the starter story (no double information).

Notice same for error() - we could again explain about logging, but we already explained in log()